### PR TITLE
Add column quotes in macro to avoid build failure

### PR DESCRIPTION
**Pull request by seanglynn-thrive**
Resolves: https://github.com/dbt-labs/dbt-adapters/issues/157

### Problem
The macro: `default__alter_relation_add_remove_columns` generated by incremental config: `on_schema_change = 'append_new_columns'` is causing our build to fail when it encounters columns that are named as reserved keywords [BigQuery SQL syntax](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#reserved_keywords).

### Solution

Wrap column reference with quotes to avoid a SQL exception being thrown.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [ ] This PR includes tests
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc)

### CODE REVIEW

**Comment by dbeatty10** [comment]
@cla-bot check

**Comment by dbeatty10** [comment]
Thanks for opening this PR @seanglynn-thrive !

I just created and added changelog entry file for this PR: `.changes/unreleased/Fixes-20230918-145127.yaml`

Could you do the following?
1. Open an associated issue here (so there is a more detailed description of the problem, including a simple code example, that we can link to it in the changelog).
1. Update `.changes/unreleased/Fixes-20230918-145127.yaml` with the resulting issue number.
1. Review each of the items in the PR checklist and check off those that have been completed already.
1. Make an attempt at checking off the rest.

In terms of adding new test cases, the problem and solution are similar to [#919](https://github.com/dbt-labs/dbt-bigquery/pull/919/files), and you can use that PR for inspiration for how to implement.

Specifically, we'll want to make sure the tests verify that both unquoted and quoted identifiers still work when appending new columns (in addition to a quoted identifier with a reserved keyword).

**Comment by seanglynn-thrive** [comment]
> Thanks for opening this PR @seanglynn-thrive !
> 
> I just created and added changelog entry file for this PR: `.changes/unreleased/Fixes-20230918-145127.yaml`
> 
> Could you do the following?
> 
> 1. Open an associated issue here (so there is a more detailed description of the problem, including a simple code example, that we can link to it in the changelog).
> 2. Update `.changes/unreleased/Fixes-20230918-145127.yaml` with the resulting issue number.
> 3. Review each of the items in the PR checklist and check off those that have been completed already.
> 4. Make an attempt at checking off the rest.
> 
> In terms of adding new test cases, the problem and solution are similar to [#919](https://github.com/dbt-labs/dbt-bigquery/pull/919/files), and you can use that PR for inspiration for how to implement.
> 
> Specifically, we'll want to make sure the tests verify that both unquoted and quoted identifiers still work when appending new columns (in addition to a quoted identifier with a reserved keyword).

@dbeatty10 
I've opened an issue here: https://github.com/dbt-labs/dbt-core/issues/8667 and updated the changelog in the latest commit. However, I'm not really sure where I can add unit tests for this change. The PR you have referenced seems to refer to pytest files that no longer exists on the latest branch :( 
Would you mind directing me to the best location to add these unit tests and I will be more than happy to add them?

File: .changes/unreleased/Fixes-20230918-145127.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Fixes
+body: Add column quotes in macro to avoid build failure
+time: 2023-09-18T14:51:27.988997-06:00
+custom:
+  Author: seanglynn-thrive
+  Issue: "6154"
```
Comment (dbeatty10) [inline .changes/unreleased/Fixes-20230918-145127.yaml:6]: Did you mean #8667 instead of #6154?

```suggestion
  Issue: "8667"
```

**Review by dbeatty10 — CHANGES_REQUESTED** [review]

**Comment by dbeatty10** [comment]
Nice @seanglynn-thrive 😎

I left a review comment within the changelog entry -- could you take a look?

[Here's](https://github.com/dbt-labs/dbt-bigquery/blob/27ade3df84b040d4140c1870a033b20c2495b5cd/tests/functional/adapter/incremental/test_incremental_on_schema_change.py#L86) the example test from https://github.com/dbt-labs/dbt-bigquery/pull/919.

Since the example above is from the dbt-bigquery repo, but your PR is within the dbt-core repo, you can use a search similar to the following to find a relevant test to modify (or copy-paste from):
- GitHub code search: [`repo:dbt-labs/dbt-core append_new_columns`](https://github.com/search?q=repo%3Adbt-labs%2Fdbt-core%20append_new_columns&type=code)

Here's some lines of code you could take a peek at:
1. [fixture model definition](https://github.com/dbt-labs/dbt-core/blob/d597b80486ebb409685fafb23d972f5361ddf7fc/tests/functional/incremental_schema_tests/fixtures.py#L287)
2. [import fixture](https://github.com/dbt-labs/dbt-core/blob/d597b80486ebb409685fafb23d972f5361ddf7fc/tests/adapter/dbt/tests/adapter/incremental/test_incremental_on_schema_change.py#L18)
3. [add the fixture to the dbt project for a pytest case](https://github.com/dbt-labs/dbt-core/blob/d597b80486ebb409685fafb23d972f5361ddf7fc/tests/adapter/dbt/tests/adapter/incremental/test_incremental_on_schema_change.py#L37)
4. [run the dbt project within a test case that pytest will pick up](https://github.com/dbt-labs/dbt-core/blob/d597b80486ebb409685fafb23d972f5361ddf7fc/tests/adapter/dbt/tests/adapter/incremental/test_incremental_on_schema_change.py#L88)

Adding / modifying tests can be tricky sometimes, so just let us know if you run into any issues and need a hand.

**Review by dbeatty10 — APPROVED** [review]

**Comment by dbeatty10** [comment]
Thanks for taking the time to open this PR @seanglynn-thrive! Since opening, we've [decoupled dbt Adapters from dbt Core](https://github.com/dbt-labs/dbt-core/discussions/9171), and this code now lives in a separate repo: [dbt-adapters](https://github.com/dbt-labs/dbt-adapters).

A consequence of the decoupling is that PR can't be merged anymore as is, so we're closing it. For more context see [#9171](https://github.com/dbt-labs/dbt-core/discussions/9171).

The linked issue has already been transferred. Please don't hesitate to re-open your proposed code changes within this PR in the dbt-adapters repo.

**Pull request closed without merging**